lib/fluent/plugin/in_opensearch.rb

```diff
@@ -239,6 +240,43 @@ def parse_time(value, event_time, tag)
       return Time.at(event_time).to_time
     end
 
+    def get_reachable_hosts(hosts=nil)
+      reachable_hosts = []
+      attempt = 0
+      loop do
+        hosts.each do |host|
+          begin
+            if @infinite_check_connection == true
+              check_host = OpenSearch::Client.new(
+                host: ["#{host[:scheme]}://#{host[:host]}:#{host[:port]}"],
+                user: host[:user],
+                password: host[:password],
+                reload_connections: true,
+                resurrect_after: @resurrect_after,
+                reload_on_failure: @reload_on_failure,
+                transport_options: { ssl: { verify: @ssl_verify, ca_file: @ca_file, version: @ssl_version } }
+              )
```

Comment:
Creating a method that is only used for verifying reachable hosts would be better instead of just writing down whole things to do so.